### PR TITLE
Overdue updates to DG

### PR DESCRIPTION
**Pull request by LeeEnHao**
## Description

Removed proposed implementation for CommandHistory -> Implementation.

Fixes #351  

## Testing
NIL

## Remarks
NIL

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -256,6 +256,32 @@ The following sequence diagram shows how the suggest operation works:
 
 ![Suggest Sequence Diagram](images/SuggestSequenceDiagram.png)
 
+### Command History
+
+A command history greatly optimise the user's productivity by removing the need to repetitively type out similar commands.
```
Comment (rtshkmr) [inline docs/DeveloperGuide.md]:
```suggestion
A command history greatly optimises the user's productivity by removing the need to repetitively type out similar commands.
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -256,6 +256,32 @@ The following sequence diagram shows how the suggest operation works:
...
+Furthermore, it allows backwards viewing of previous commands given, which is similar to the CLI experience. However, the difference between the traditional CLI history and TBM CLI history is that TBM's history only accepts valid commands. Invalid commands entered will not be included will not sppear in the command history.
```
Comment (rtshkmr) [inline docs/DeveloperGuide.md]:
```suggestion
Furthermore, it allows backwards viewing of previous commands given, which is similar to the CLI experience. However, the difference between the traditional CLI history and TBM CLI history is that TBM's history only accepts valid commands. Invalid commands entered will not be included will not appear in the command history.
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -256,6 +256,32 @@ The following sequence diagram shows how the suggest operation works:
...
+The modification of previous commands in the history do not override the current command as well. The current input being edited is stored in a separate variable was the user scrolls through the command history.
```
Comment (rtshkmr) [inline docs/DeveloperGuide.md]: This sentence is a little confusing man 😟 

"The modification of previous commands ..." 
and 

also typo on the "was"  --> "as" probably

**Review by rtshkmr — CHANGES_REQUESTED** [review]
Suggested some nits to look into

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -256,6 +256,32 @@ The following sequence diagram shows how the suggest operation works:
...
+
+A command history greatly optimises the user's productivity by removing the need to repetitively type out similar commands.
+Furthermore, it allows backwards viewing of previous commands given, which is similar to the CLI experience. 
+
+However, the difference between the traditional CLI history and TBM CLI history is that TBM's history only accepts valid commands. Invalid commands entered will not be included will not appear in the command history.
+Furthermore, the modification of previous commands in the history do not override the current command as well. The current input being edited is stored in a separate variable as the user scrolls through the command history. This is proposed upgrade for the CLI history feature.
+
+The implementation of command history is backed by a list of strings `history`. Upon the entering of a valid command, the command history, `CommandHistory`.
+
+![User Command History Activity](images/CommandHistoryActivity.png))
```
Comment (raysonkoh) [inline docs/DeveloperGuide.md]: Extra closing paren at end of this line

**Review by raysonkoh — APPROVED** [review]
LGTM

**Review by rtshkmr — APPROVED** [review]
LGTM

**Pull request merged**